addons/sale_mrp/models/account_move.py

```diff
@@ -15,10 +15,13 @@ def _get_cogs_value(self):
             # If there are changes in BOMs between the stock moves creation and the
             # invoice validation a wrong price will be taken
             boms = so_line.move_ids.filtered(lambda m: m.state != 'cancel').mapped('bom_line_id.bom_id').filtered(lambda b: b.type == 'phantom')
-            if boms:
+            bom_dropshipped_kit = self.product_id not in self.sale_line_ids.move_ids.mapped("product_id") and \
+                all(move._is_dropshipped() for move in self.sale_line_ids.move_ids) and \
+                not boms
+            if boms or bom_dropshipped_kit:
```

Comment:
```suggestion
            moves = so_line.move_ids.filtered(lambda m: m.state != 'cancel')
            boms = moves.bom_line_id.bom_id'.filtered(lambda b: b.type == 'phantom')
            is_bom_dropshipped_kit = not boms and \
                self.product_id not in moves.product_id and \
                all(move._is_dropshipped() for move in moves)
            if boms or is_bom_dropshipped_kit:
```
A few small things:
- I think the check on the moves being `cancel` should still apply
- You don't need to use `mapped()` when going through recordsets
- Always put fastest conditions first. If there is something in `boms`, we don't need to check anything else, so it saves some time